parser-typechecker/src/Unison/TermPrinter.hs

```diff
@@ -1087,3 +1100,78 @@ unLetBlock t = rec t where
         Just (innerBindings, innerBody) | dontIntersect bindings innerBindings ->
           Just (bindings ++ innerBindings, innerBody)
         _ -> Just (bindings, body)
+
+pattern LamsNamedMatch' vs branches <- (unLamsMatch' -> Just (vs, branches))
+
+-- This function is used to detect places where lambda case syntax can be used.
+-- When given lambdas of a form that corresponds to a lambda case, it returns
+-- `Just (varsBeforeCases, branches)`. Leading vars are the vars that should be
+-- shown to the left of the `-> cases`.
+--
+-- For instance, if given this term:
+--
+--   x y z -> match z with
+--     [] -> "empty"
+--     (h +: t) -> "nonempty"
+--
+-- this function will return Just ([x,y], [[] -> "empty", (h +: t) -> "nonempty"])
+-- and it would be rendered as `x y -> cases [] -> "empty"`
```

Comment:
missing second branch in example in comment

Reply:
Thanks for spotting! fixed in latest commit